src/Kestrel.Core/Internal/Http/Http1Connection.cs

```diff
@@ -454,10 +469,8 @@ protected override bool TryParseRequest(ReadResult result, out bool endConnectio
             {
                 if (_requestProcessingStatus == RequestProcessingStatus.ParsingHeaders)
                 {
-                    BadHttpRequestException.Throw(RequestRejectionReason
-                        .MalformedRequestInvalidHeaders);
+                    BadHttpRequestException.Throw(RequestRejectionReason.MalformedRequestInvalidHeaders);
                 }
-                throw;
```

Comment:
This seems suspicious. Why is this ok? We always want to throw even in the else case?

Reply:
Aha! Yes, will add back

Reply:
Fixed